frontend/src/core/alerts/state.ts

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { PackageInstallationStatus } from "@/core/kernel/messages";
+import { createReducer } from "@/utils/createReducer";
+import { atom, useAtomValue, useSetAtom } from "jotai";
+import { useMemo } from "react";
+
+export interface MissingPackageAlert {
+  id: string;
+  kind: "missing";
+  packages: string[];
+  isolated: boolean;
+}
+
+export interface InstallingPackageAlert {
+  id: string;
+  kind: "installing";
+  packages: PackageInstallationStatus;
+}
+
+export function isMissingPackageAlert(
+  alert: MissingPackageAlert | InstallingPackageAlert,
+): alert is MissingPackageAlert {
+  return alert.kind === "missing";
+}
+
+export function isInstallingPackageAlert(
+  alert: MissingPackageAlert | InstallingPackageAlert,
+): alert is InstallingPackageAlert {
+  return alert.kind === "installing";
+}
+
+/** Prominent alerts.
+ *
+ * Right now we only have one type of alert.
+ */
+interface AlertState {
+  packageAlert: MissingPackageAlert | InstallingPackageAlert | null;
+}
+
+const { reducer, createActions } = createReducer(
+  () => ({ packageAlert: null }) as AlertState,
+  {
+    addPackageAlert: (
```

Comment:
i think you should create a shared typed `Identified` that extends `MissingPackageAlert`. This way you can share it with the message type too

```ts
type Identified<T> =  {id: string} & T

export interface MissingPackageAlert {
  kind: "missing";
  packages: string[];
  isolated: boolean;
}

export interface InstallingPackageAlert {
  kind: "installing";
  packages: PackageInstallationStatus;
}
```

```ts
interface AlertState {
  packageAlert: Identified<MissingPackageAlert> | Identified<InstallingPackageAlert> | null;
}
```

and in most cases, you just need `MissingPackageAlert`

also, i would create the uuid in the reducer (better for more logic there since its more testable than react components.

```ts
      return {
        ...state,
        packageAlert: {id: generatedUUID(), ...alert}
      };
```